src/placement/jump_map.c

```diff
@@ -248,6 +284,51 @@ pl_map2jmap(struct pl_map *map)
 	return container_of(map, struct pl_jump_map, jmp_map);
 }
 
+static inline uint32_t
+get_num_domains(struct pool_domain *curr_dom, enum PL_OP_TYPE op_type)
+{
+	struct pool_domain *next_dom;
+	struct pool_target *next_target;
+	uint32_t num_dom;
+	uint32_t fseq;
+	uint8_t status;
+
+	if (curr_dom->do_children == NULL)
+		num_dom = curr_dom->do_target_nr;
+	else
+		num_dom = curr_dom->do_child_nr;
+
+	if (op_type == PL_ADD)
+		return num_dom;
+
+	if (curr_dom->do_children != NULL) {
+		next_dom = &curr_dom->do_children[num_dom - 1];
+		fseq = next_dom->do_comp.co_fseq;
+		status = next_dom->do_comp.co_status;
+
+		while (num_dom - 1 > 0 && fseq == 1 &&
+		       status == PO_COMP_ST_NEW) {
```

Comment:
This is a bit confusing. so it is trying to find out all number of doms, except the NEW domain? what does this fseq == 1 means here?

Reply:
Yes, because for jump consistent hash to produce the original layout we need to know the original number of targets, otherwise it won't match and we'll get the new layout.

I actually don't think we need the fseq comparison any more. We all originally talked about needing it to determine which targets were being added but it ended up being unnecessary.

Reply:
so you will remove it in the following patch?

Reply:
I was going to push a patch to fix it now. However it might be a good idea to include it in the migration/rebuild patch that will follow this one so we can land this one sooner.